### PR TITLE
Complete mod rewrite

### PR DESCRIPTION
**Pull request by senseiwells**
- Fixes numerous bugs
- Removed a shit ton of junk code
- Reworked and refactored a shit ton of code

### CODE REVIEW

**Comment by senseiwells** [comment]
- Reworked Config system, now much more cleaner and much easier to implement new configs when needed
- Fixed EnumArgumentType not being serialized to the client (Just psuedoing it as a string)
- Competely rewrote reloadFakePlayer, so it uses Jsons and retains tick info about actions
- Removed a huge chunk of the Logger code that was unnecessary
- Fixed force game mode kicking cameraMode
- Added `essentialCarefulDrop` which puts entity drops straight into the player's inventory when crouching
- Reworked subscribe system so it's more modular
- General code cleanup

File: src/main/java/essentialaddons/commands/CommandRename.java [hunk 1]
```diff
@@ -14,15 +14,16 @@
...
-                                itemStack.setCustomName(new LiteralText(context.getArgument("name", String.class)));
-                            return 0;
-                        })
-                )
+        dispatcher.register(literal("rename").requires((player) -> SettingsManager.canUseCommand(player, EssentialSettings.commandRename))
+            .then(argument("name", StringArgumentType.string())
+                .executes(context -> {
+                    ItemStack itemStack = context.getSource().getPlayer().getMainHandStack();
+                    if (itemStack.getItem() != Items.AIR) {
+                        itemStack.setCustomName(new LiteralText(context.getArgument("name", String.class)));
```
Comment (Super-Santa) [inline src/main/java/essentialaddons/commands/CommandRename.java:22]: Would be nice to have error and success messages here.

**Comment by senseiwells** [comment]
I will finish this at some point xD

**Comment by senseiwells** [comment]
Ok this should be ready for merging now

**Pull request merged**